graphql_compiler/schema_generation/sqlalchemy/edge_descriptors.py

```diff
@@ -56,3 +57,54 @@ def validate_edge_descriptors(vertex_name_to_table, direct_edges):
                 raise InvalidSQLEdgeError('SQL edge {} with edge descriptor {} references a '
                                           'non-existent column {}'
                                           .format(edge_name, direct_edge_descriptor, column_name))
+
+
+def generate_direct_edge_descriptors_from_foreign_keys(vertex_name_to_table):
+    """Return a set of DirectEdgeDescriptor objects from the foreign keys in the SQLAlchemy tables.
+
+    Args:
+        vertex_name_to_table: Dict[str, Table], a mapping of vertex type names to the underlying
+                              SQLAlchemy table objects.
+
+    Return:
+        set of DirectEdgeDescriptor objects extrapolated from the foreign keys.
+    """
+    table_to_vertex_name = _get_table_to_vertex_name(vertex_name_to_table)
+    direct_edge_descriptors = set()
+
+    number_of_composite_foreign_keys = 0
+    for vertex_name, table in vertex_name_to_table.items():
+        for fk_constraint in table.foreign_key_constraints:
+            foreign_key_columns = fk_constraint.columns
+            # The .elements attribute refers to a list of ForeignKey objects.
+            referenced_columns = [element.column for element in fk_constraint.elements]
+            if len(foreign_key_columns) != 1 or len(referenced_columns) != 1:
+                number_of_composite_foreign_keys += 1
+            else:
+                foreign_key_column = next(iter(foreign_key_columns))
+                referenced_column = next(iter(referenced_columns))
+                referenced_table = referenced_column.table
+                referenced_vertex_name = table_to_vertex_name[referenced_table]
+                direct_edge_descriptors.add(
+                    DirectEdgeDescriptor(
+                        from_vertex=vertex_name,
+                        from_column=foreign_key_column.name,
+                        to_vertex=referenced_vertex_name,
+                        to_column=referenced_column.name
+                    ))
+
+    if number_of_composite_foreign_keys:
+        warnings.warn('Ignored {} edges implied by composite foreign keys. We currently do not '
+                      'support SQL edges with multiple source/destination columns.')
+
+    return direct_edge_descriptors
+
+
+def _get_table_to_vertex_name(vertex_name_to_table):
+    """Return a mapping of SQLAlchemy Table objects to their vertex type names."""
+    table_to_vertex_name = {}
+
+    for vertex_name, table in vertex_name_to_table.items():
+        table_to_vertex_name[table] = vertex_name
```

Comment:
Assert that this line doesn't accidentally overwrite something? It shouldn't, but better safe than sorry.